packages/core-plugin-api/src/extensions/extensions.tsx

```diff
@@ -37,6 +38,7 @@ export function createRoutableExtension<
 >(options: {
   component: () => Promise<T>;
   mountPoint: RouteRef;
+  name?: string;
```

Comment:
It turns out that the lazy-loading logic in here means that when `createReactExtension()` is ultimately called, the underlying component never gets properly named (because `displayName` and `name` are never set on a lazy-loaded object).  ...All of these changes were my way of giving plugin developers a way to explicitly set a component name at extension registration-time.

Reply:
To be honest we're a bit unsure about the value of the component name in the analytics. It seems something that is a bit implementation specific and also might change over time in unexpected ways. How do you feel about leaving that piece out for new? :grin:

Reply:
Yeah, lemme talk through motivations real quick, and then I'd love to hear your thoughts:

Internally, I've found it very handy to be able to slice and dice events at a variety of levels of detail.  

Sometimes you just want to take a really shallow look at aggregate events of all kinds happening under a plugin.  For example, a PM might ask, _"What's the Catalog MAU?"_ -- and one could answer with... `SELECT DISTINCT COUNT(userId) WHERE pluginId='catalog' AND date >= [30 Days Ago]`.

Other times, you need to be able to distinguish between two otherwise identical events (e.g. same verb/action + noun/subject) where the only difference is the immediate context of the interaction on the page.  For example, a PM might ask, _"Do people on Catalog Entity pages click to the entity's owner more by clicking the link in the About card, or in the header?"_  -- which you might be able to answer with...  `SELECT componentName, COUNT(*) WHERE verb='click' AND noun LIKE '/catalog/default/group/%' AND routeRef='Entity' GROUP BY componentName`.

When the App Integrator asking the questions and the Plugin Developer implementing the instrumentation are one in the same, you have the luxury of being able to just add extra contextual metadata to the events to be able to distinguish them, but then here we are with Backstage!  So having those levels of detail (what plugin? what page? what part of the page?) instrumented out-of-the-box is handy.

My hope was to have a system such that, just by building your plugin "the right way" (making use of extensions / the composability API / etc.), you would get sufficient levels of detail to handle most such cases.  ...Only after playing around with it, did I realize that component name was kind of broken (basically everything is just `Component`).

Any other ideas on how to achieve the design goal above?

Reply:
What I'm thinking is that it's the "extension name" that we want rather than the component name, which tbh is pretty much exactly this just that I thought about it from the other direction 😅

It's a lil bit unfortunate if we have to repeat the name for every extension declaration. I'm thinking we might be able to just grab the display name of the extension component itself as we traverse the element tree. Prolly works well with the plugin and route gathering as well?

Reply:
Alright, I think by just renaming `componentName` to `extension`, we get where we want.  See top-level comment (forthcoming) on this PR for full details.